KWZP2022/KWZP2022/SkladPolproduktForm.cs

```diff
@@ -0,0 +1,134 @@
+        }
+
+        private void dgvPolprodukt_CellContentClick(object sender, DataGridViewCellEventArgs e)
+        {
+            txtPolprodukt.Text = dgvPolprodukt.CurrentRow.Cells[1].Value.ToString();
+        }
+
+        private void dgvMaterial_CellContentClick(object sender, DataGridViewCellEventArgs e)
+        {
+            txtMaterial.Text = dgvMaterial.CurrentRow.Cells[2].Value.ToString();
+        }
+        private void dgvSkladPolprodukt_CellContentClick(object sender, DataGridViewCellEventArgs e)
+        {
+            txtPolprodukt.Text = dgvSkladPolprodukt.CurrentRow.Cells[1].Value.ToString();
+            txtMaterial.Text = dgvSkladPolprodukt.CurrentRow.Cells[2].Value.ToString();
+            txtIlosc.Text = dgvSkladPolprodukt.CurrentRow.Cells[4].Value.ToString();
+        }
+
+        private void btnDodaj_Click(object sender, EventArgs e)
+        {
+            if (String.IsNullOrEmpty(txtPolprodukt.Text) || String.IsNullOrEmpty(txtMaterial.Text) || String.IsNullOrEmpty(txtIlosc.Text))
+            {
+                MessageBox.Show("Uzupełnij brakujące informacje.");
+                
+            }
+            else
+            {
+                Sklad_polprodukt skladPolprodukt = new Sklad_polprodukt();
+                skladPolprodukt.ID_polprodukt = int.Parse(dgvPolprodukt.CurrentRow.Cells[0].Value.ToString());
+                skladPolprodukt.ID_material = int.Parse(dgvMaterial.CurrentRow.Cells[0].Value.ToString());
+                skladPolprodukt.Liczba = int.Parse(txtIlosc.Text);
+
+                db.Sklad_polprodukt.Add(skladPolprodukt);
+                db.SaveChanges();
+                refreshScreen();
+            }
+
+        }
+
+        private void btnUsun_Click(object sender, EventArgs e)
+        {
+            DialogResult deleteResult = MessageBox.Show("Czy na pewno chcesz usunąć powiązanie między półproduktem '" + dgvSkladPolprodukt.CurrentRow.Cells[1].Value.ToString() + "',a materiałem '"+ dgvSkladPolprodukt.CurrentRow.Cells[2].Value.ToString() + "'?", "Question",MessageBoxButtons.YesNo);
+            if (deleteResult == DialogResult.Yes)
+            {
+                int currentSkladPolprodukt = int.Parse(dgvSkladPolprodukt.CurrentRow.Cells[0].Value.ToString());
+
+                db.Sklad_polprodukt.Remove(db.Sklad_polprodukt.Where(polprodukt => polprodukt.ID_sklad_polprodukt == currentSkladPolprodukt).First());
+                db.SaveChanges();
+                refreshScreen();
+
+            }
+        }
+
+        private void btnOdswiez_Click(object sender, EventArgs e)
+        {
+            refreshScreen();
+        }
+
+
+        private void btnAktualizuj_Click(object sender, EventArgs e)
+        {
+            int currentID = int.Parse(dgvSkladPolprodukt.CurrentRow.Cells[0].Value.ToString());
+            int newValue = int.Parse(txtIlosc.Text);
+        
+            var result = db.Sklad_polprodukt.SingleOrDefault(b => b.ID_sklad_polprodukt == currentID);
```

Comment:
proszę o właściwy typ

Reply:
Tutaj nie wiem jakim innym typem mógłbym to zastąpić.
P.S. po zaakceptowaniu dotychaczowych zmian przystąpię do rozwiązania konfliktów.